### PR TITLE
Missions screen

### PR DESCRIPTION
**Pull request by tddumont**
This PR adds a missions screen, which will serve as the new home screen. All the missions in a course are retrieved from the backend and displayed in a list. When a mission is selected, a mission overview card is displayed. The mission overview can be used to navigate to the individual mission page.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -158,6 +159,7 @@ function App() {
           <Route component = {Mission} exact path = '/mission'/>
```
Comment (ryanmadamba) [inline src/App.js:183]: This component makes sense as the MissionsScreen. We should probably make more of a distinction between Mission and MissionsScreen.

**Review by ryanmadamba — APPROVED** [review]
Looks great Ted!

File: src/App.js [hunk 2]
```diff
@@ -6,14 +6,15 @@ import {Route, Switch} from 'react-router-dom';
...
 import GradeScreen from './Components/GradeScreen';
+import MissionsScreen from './Components/MissionsScreen';
 import Mission from './Components/Mission';
 import Task from './Components/Task';
 import MTaskOverview from './Components/MTaskOverview';
 import NavDrawer from './Components/NavDrawer';
 import { onError } from '@apollo/client/link/error';
 import "./App.css";
 
-const HOME_SCREEN_PATH = 'mission';
```
Comment (Bart-Ilg) [inline src/App.js:16]: name fix

File: src/App.js [hunk 3]
```diff
@@ -167,8 +168,8 @@ function App() {
           <li><a onClick={() => Auth.federatedSignIn()}>Sign In</a></li> :
     
           <div>
-          <NavDrawer className="drawer"/>
-          <li><a onClick={() => Auth.signOut()}>Sign Out</a></li>          
+            <NavDrawer className="drawer"/>
```
Comment (Bart-Ilg) [inline src/App.js:171]: good to improve indentation

File: src/Components/Mission.js [hunk 4]
```diff
@@ -5,9 +5,11 @@ import MTaskOverview from './MTaskOverview.js';
 import './Mission.css';
 
 //This component is used to display the mission page.
-function Mission() {
+function Mission(props) {
```
Comment (Bart-Ilg) [inline src/Components/Mission.js:8]: good to add props support, more versatile

File: src/Components/Mission.js [hunk 5]
```diff
@@ -5,9 +5,11 @@ import MTaskOverview from './MTaskOverview.js';
...
+  const missionId = props?.location?.state?.id;
+
   const { loading, error, data, refetch} = useQuery(GET_MISSION, {
-    variables: { id: "da0719ba103" },
```
Comment (Bart-Ilg) [inline src/Components/Mission.js:10]: good, more descriptive

File: src/Components/MissionsScreen.css [hunk 6]
```diff
@@ -0,0 +1,57 @@
+.card {
```
Comment (Bart-Ilg) [inline src/Components/MissionsScreen.css:1]: clean and well formatted

File: src/Components/Task.js [hunk 7]
```diff
@@ -123,7 +123,12 @@ function Task(props) {
 
   const submitTask = () => {
     alert(`${title} submitted.`);
-    props.history.push('/');
+    props.history.push({
```
Comment (Bart-Ilg) [inline src/Components/Task.js:126]: good, documenting info instead of placeholder

File: src/gqlQueries.js [hunk 8]
```diff
@@ -141,6 +141,16 @@ export const SAVE_FRQUESTION = gql`
   }
 `;
 
+export const GET_MISSIONS = gql`
```
Comment (Bart-Ilg) [inline src/gqlQueries.js:144]: good query, maybe more semicolon to line above it?

**Review by Bart-Ilg — APPROVED** [review]
nicely formatted and doesnt seem like it has excessive fluff, good work

**Pull request merged**